### DISCUSSION TITLE
How do I change the type of column?

### DISCUSSION THREAD

**Discussion by EvanGllt** [Budibase/budibase#14158] [Help] [answered]
Hello, there! 👋

I have a problem with a tinyint, but I can't edit the column type:

![aide 1](https://github.com/user-attachments/assets/1d3df1fb-79a7-4ee5-858a-ed59080d86d3)

I use an MS SQL Server database.

Do you know why? 

Thanks in advance.

**Comment by mikesealey** [comment, marked as answer]
Hi @EvanGllt 

If you're using an external data source your columns can be set in the schema of the query you used to create them.

Data Tab >> [ Data Source ] >> [ Query ] >> Run Query >> Schema
![image](https://github.com/user-attachments/assets/af5bd5c8-8187-4544-90eb-37f2751b9355)

Hope this helps!

**Reply by EvanGllt** [reply]
Thank you for your reply.

I hadn't seen that for custom queries.

However, is it possible to also change the type on raw arrays?

![aide 1](https://github.com/user-attachments/assets/fc3ac1bb-912f-4503-8f8d-65ee9957b6d5)
